apps/hash-graph/libs/graph/src/store/postgres/knowledge/entity/mod.rs

```diff
@@ -457,275 +467,255 @@ where
+                    }))
+                    .map(|relation_and_subject| (entity_id, relation_and_subject)),
+            );
+            if relationships.len() == current_num_relationships {
+                return Err(Report::new(InsertionError)
+                    .attach_printable("At least one relationship must be provided"));
+            }
         }
 
-        for entity_type_id in &params.entity_type_ids {
-            let entity_type_id = EntityTypeId::from_url(entity_type_id);
-            self.authorization_api
-                .check_entity_type_permission(
-                    actor_id,
-                    EntityTypePermission::Instantiate,
-                    entity_type_id,
-                    Consistency::FullyConsistent,
-                )
-                .await
-                .change_context(InsertionError)?
-                .assert_permission()
-                .change_context(InsertionError)
-                .attach(StatusCode::PermissionDenied)?;
+        let (instantiate_permissions, zookie) = self
+            .authorization_api
+            .check_entity_types_permission(
+                actor_id,
+                EntityTypePermission::Instantiate,
+                entity_type_ids,
+                Consistency::FullyConsistent,
+            )
+            .await
+            .change_context(InsertionError)?;
+        if instantiate_permissions.values().any(Not::not) {
+            return Err(Report::new(InsertionError)
+                .attach(StatusCode::PermissionDenied)
+                .attach_printable(
+                    "The actor does not have permission to instantiate one or more entity types",
+                ));
         }
 
-        if Some(params.owned_by_id.into_uuid()) != params.entity_uuid.map(EntityUuid::into_uuid) {
-            self.authorization_api
-                .check_web_permission(
+        if !checked_web_ids.is_empty() {
+            let (create_entity_permissions, _zookie) = self
+                .authorization_api
+                .check_webs_permission(
                     actor_id,
                     WebPermission::CreateEntity,
-                    params.owned_by_id,
-                    Consistency::FullyConsistent,
+                    checked_web_ids,
+                    Consistency::AtLeastAsFresh(&zookie),
                 )
                 .await
-                .change_context(InsertionError)?
-                .assert_permission()
-                .change_context(InsertionError)
-                .attach(StatusCode::PermissionDenied)?;
+                .change_context(InsertionError)?;
+            if create_entity_permissions.values().any(Not::not) {
+                return Err(Report::new(InsertionError)
+                    .attach(StatusCode::PermissionDenied)
+                    .attach_printable(
+                        "The actor does not have permission to create entities for one or more \
+                         web ids",
```

Comment:
Same question as types, will the webs denied be in the error somewhere?

Reply:
No, will add them, good flag!

Reply:
Fixed in [`bbcdbe0` (#4398)](https://github.com/hashintel/hash/pull/4398/commits/bbcdbe0c5bf1fd11393bd375684d1db8d91d6fe3)